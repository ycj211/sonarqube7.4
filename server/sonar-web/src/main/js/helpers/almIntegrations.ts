/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { isLoggedIn } from './users';
import { CurrentUser, AlmOrganization } from '../app/types';

export function hasAdvancedALMIntegration(user: CurrentUser) {
  return (
    isLoggedIn(user) && (isBitbucket(user.externalProvider) || isGithub(user.externalProvider))
  );
}

export function isBitbucket(almKey?: string) {
  return almKey && almKey.startsWith('bitbucket');
}

export function isGithub(almKey?: string) {
  return almKey === 'github';
}

export function isVSTS(almKey?: string) {
  return almKey === 'microsoft';
}

export function isPersonal(organization?: AlmOrganization) {
  return Boolean(organization && organization.personal);
}

export function sanitizeAlmId(almKey?: string) {
  if (isBitbucket(almKey)) {
    return 'bitbucket';
  }
  return almKey;
}
