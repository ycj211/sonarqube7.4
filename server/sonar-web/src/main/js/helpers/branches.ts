/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { sortBy } from 'lodash';
import {
  BranchLike,
  Branch,
  BranchType,
  ShortLivingBranch,
  LongLivingBranch,
  PullRequest,
  MainBranch,
  BranchParameters
} from '../app/types';

export function isBranch(branchLike?: BranchLike): branchLike is Branch {
  return branchLike !== undefined && (branchLike as Branch).isMain !== undefined;
}

export function isShortLivingBranch(branchLike?: BranchLike): branchLike is ShortLivingBranch {
  return (
    isBranch(branchLike) &&
    !branchLike.isMain &&
    (branchLike as ShortLivingBranch).type === BranchType.SHORT
  );
}

export function isLongLivingBranch(branchLike?: BranchLike): branchLike is LongLivingBranch {
  return (
    isBranch(branchLike) &&
    !branchLike.isMain &&
    (branchLike as LongLivingBranch).type === BranchType.LONG
  );
}

export function isMainBranch(branchLike?: BranchLike): branchLike is MainBranch {
  return isBranch(branchLike) && branchLike.isMain;
}

export function isPullRequest(branchLike?: BranchLike): branchLike is PullRequest {
  return branchLike !== undefined && (branchLike as PullRequest).key !== undefined;
}

export function getPullRequestDisplayName(pullRequest: PullRequest) {
  return `${pullRequest.key} â€“ ${pullRequest.title}`;
}

export function getBranchLikeDisplayName(branchLike: BranchLike) {
  return isPullRequest(branchLike) ? getPullRequestDisplayName(branchLike) : branchLike.name;
}

export function getBranchLikeKey(branchLike: BranchLike) {
  return isPullRequest(branchLike) ? `pull-request-${branchLike.key}` : `branch-${branchLike.name}`;
}

export function getBranchQualityGateColor(status: string) {
  let indicatorColor = 'gray';
  if (status === 'ERROR') {
    indicatorColor = 'red';
  } else if (status === 'WARN') {
    indicatorColor = 'orange';
  } else if (status === 'OK') {
    indicatorColor = 'green';
  }
  return indicatorColor;
}

export function isSameBranchLike(a: BranchLike | undefined, b: BranchLike | undefined) {
  // main branches are always equal
  if (isMainBranch(a) && isMainBranch(b)) {
    return true;
  }

  // short- and long-living branches are compared by type and name
  if (
    (isLongLivingBranch(a) && isLongLivingBranch(b)) ||
    (isShortLivingBranch(a) && isShortLivingBranch(b))
  ) {
    return a.type === b.type && a.name === b.name;
  }

  // pull requests are compared by id
  if (isPullRequest(a) && isPullRequest(b)) {
    return a.key === b.key;
  }

  // finally if both parameters are `undefined`, consider them equal
  return a === b;
}

export function sortBranchesAsTree(branchLikes: BranchLike[]) {
  const result: BranchLike[] = [];

  const mainBranch = branchLikes.find(isMainBranch);
  const longLivingBranches = branchLikes.filter(isLongLivingBranch);
  const shortLivingBranches = branchLikes.filter(isShortLivingBranch);
  const pullRequests = branchLikes.filter(isPullRequest);

  // main branch is always first
  if (mainBranch) {
    result.push(
      mainBranch,
      ...getPullRequests(mainBranch.name),
      ...getNestedShortLivingBranches(mainBranch.name)
    );
  }

  // then all long-living branches
  sortBy(longLivingBranches, 'name').forEach(longLivingBranch => {
    result.push(
      longLivingBranch,
      ...getPullRequests(longLivingBranch.name),
      ...getNestedShortLivingBranches(longLivingBranch.name)
    );
  });

  // finally all orhpan pull requests and branches
  result.push(
    ...sortBy(pullRequests.filter(pr => pr.isOrphan), pullRequest => pullRequest.key),
    ...sortBy(shortLivingBranches.filter(branch => branch.isOrphan), branch => branch.name)
  );

  return result;

  /** Get all short-living branches (possibly nested) which should be merged to a given branch */
  function getNestedShortLivingBranches(mergeBranch: string) {
    const found: ShortLivingBranch[] = shortLivingBranches.filter(
      branch => branch.mergeBranch === mergeBranch
    );

    let i = 0;
    while (i < found.length) {
      const current = found[i];
      found.push(...shortLivingBranches.filter(branch => branch.mergeBranch === current.name));
      i++;
    }

    return sortBy(found, branch => branch.name);
  }

  function getPullRequests(base: string) {
    return sortBy(pullRequests.filter(pr => pr.base === base), pullRequest => pullRequest.key);
  }
}

export function getBranchLikeQuery(branchLike?: BranchLike): BranchParameters {
  if (isShortLivingBranch(branchLike) || isLongLivingBranch(branchLike)) {
    return { branch: branchLike.name };
  } else if (isPullRequest(branchLike)) {
    return { pullRequest: branchLike.key };
  } else {
    return {};
  }
}

// Create branch object from branch name or pull request key
export function fillBranchLike(
  branch?: string,
  pullRequest?: string
): ShortLivingBranch | PullRequest | undefined {
  if (branch) {
    return {
      isMain: false,
      mergeBranch: '',
      name: branch,
      type: BranchType.SHORT
    } as ShortLivingBranch;
  } else if (pullRequest) {
    return { base: '', branch: '', key: pullRequest, title: '' } as PullRequest;
  }
  return undefined;
}
