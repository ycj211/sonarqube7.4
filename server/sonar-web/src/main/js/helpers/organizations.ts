/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { isLoggedIn } from './users';
import { Organization, OrganizationSubscription, CurrentUser } from '../app/types';

export function isPaidOrganization(organization: Organization | undefined): boolean {
  return Boolean(organization && organization.subscription === OrganizationSubscription.Paid);
}

export function hasPrivateAccess(
  currentUser: CurrentUser,
  organization: Organization | undefined,
  userOrganizations: Organization[]
): boolean {
  return (
    !isPaidOrganization(organization) ||
    isCurrentUserMemberOf(currentUser, organization, userOrganizations)
  );
}

export function isCurrentUserMemberOf(
  currentUser: CurrentUser,
  organization: Organization | undefined,
  userOrganizations: Organization[]
): boolean {
  return Boolean(
    organization &&
      isLoggedIn(currentUser) &&
      ((organization.actions && organization.actions.admin) ||
        userOrganizations.some(org => org.key === organization.key))
  );
}
