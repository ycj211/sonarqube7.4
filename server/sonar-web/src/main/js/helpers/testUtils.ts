/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow, ShallowRendererProps, ShallowWrapper, ReactWrapper } from 'enzyme';
import { InjectedRouter } from 'react-router';
import { IntlProvider } from 'react-intl';

export const mockEvent = {
  target: { blur() {} },
  currentTarget: { blur() {} },
  preventDefault() {},
  stopPropagation() {}
};

export function click(element: ShallowWrapper | ReactWrapper, event = {}): void {
  // `type()` returns a component constructor for a composite element and string for DOM nodes
  if (typeof element.type() === 'function') {
    element.prop<Function>('onClick')();
    // TODO find out if `root` is a public api
    // https://github.com/airbnb/enzyme/blob/master/packages/enzyme/src/ReactWrapper.js#L109
    (element as any).root().update();
  } else {
    element.simulate('click', { ...mockEvent, ...event });
  }
}

export function clickOutside(event = {}): void {
  const dispatchedEvent = new MouseEvent('click', event);
  window.dispatchEvent(dispatchedEvent);
}

export function submit(element: ShallowWrapper | ReactWrapper): void {
  element.simulate('submit', {
    preventDefault() {}
  });
}

export function change(element: ShallowWrapper | ReactWrapper, value: string, event = {}): void {
  // `type()` returns a component constructor for a composite element and string for DOM nodes
  if (typeof element.type() === 'function') {
    element.prop<Function>('onChange')(value);
    // TODO find out if `root` is a public api
    // https://github.com/airbnb/enzyme/blob/master/packages/enzyme/src/ReactWrapper.js#L109
    (element as any).root().update();
  } else {
    element.simulate('change', {
      target: { value },
      currentTarget: { value },
      ...event
    });
  }
}

export function keydown(keyCode: number): void {
  const event = new KeyboardEvent('keydown', { keyCode } as KeyboardEventInit);
  document.dispatchEvent(event);
}

export function elementKeydown(element: ShallowWrapper, keyCode: number): void {
  const event = {
    currentTarget: { element },
    keyCode,
    preventDefault() {}
  };

  if (typeof element.type() === 'string') {
    // `type()` is string for native dom elements
    element.simulate('keydown', event);
  } else {
    element.prop<Function>('onKeyDown')(event);
  }
}

export function resizeWindowTo(width?: number, height?: number) {
  // `document.body.clientWidth/clientHeight` are getters by default, so we need to redefine them
  // pass `configurable: true` to allow to redefine the properties multiple times
  if (width) {
    Object.defineProperty(document.body, 'clientWidth', { configurable: true, value: width });
  }
  if (height) {
    Object.defineProperty(document.body, 'clientHeight', { configurable: true, value: height });
  }

  const resizeEvent = new Event('resize');
  window.dispatchEvent(resizeEvent);
}

export function doAsync(fn?: Function): Promise<void> {
  return new Promise(resolve => {
    setImmediate(() => {
      if (fn) {
        fn();
      }
      resolve();
    });
  });
}

// Create the IntlProvider to retrieve context for wrapping around.
const intlProvider = new IntlProvider({ locale: 'en' }, {});
const { intl } = intlProvider.getChildContext();
export function shallowWithIntl<C extends React.Component>(
  node: React.ReactElement<any>,
  options: ShallowRendererProps = {}
) {
  return shallow<C>(node, { ...options, context: { intl, ...options.context } });
}

export async function waitAndUpdate(wrapper: ShallowWrapper<any, any> | ReactWrapper<any, any>) {
  await new Promise(setImmediate);
  wrapper.update();
}

export function mockRouter(overrides: { push?: Function; replace?: Function } = {}) {
  return {
    createHref: jest.fn(),
    createPath: jest.fn(),
    go: jest.fn(),
    goBack: jest.fn(),
    goForward: jest.fn(),
    isActive: jest.fn(),
    push: jest.fn(),
    replace: jest.fn(),
    setRouteLeaveHook: jest.fn(),
    ...overrides
  } as InjectedRouter;
}
