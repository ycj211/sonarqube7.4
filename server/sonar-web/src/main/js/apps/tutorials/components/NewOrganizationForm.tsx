/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { debounce } from 'lodash';
import {
  createOrganization,
  deleteOrganization,
  getOrganization
} from '../../../api/organizations';
import AlertErrorIcon from '../../../components/icons-components/AlertErrorIcon';
import { DeleteButton, SubmitButton } from '../../../components/ui/buttons';
import { translate } from '../../../helpers/l10n';

interface Props {
  onDelete: () => void;
  onDone: (organization: string) => void;
  organization?: string;
}

interface State {
  done: boolean;
  loading: boolean;
  organization: string;
  unique: boolean;
}

export default class NewOrganizationForm extends React.PureComponent<Props, State> {
  mounted = false;

  constructor(props: Props) {
    super(props);
    this.state = {
      done: props.organization != null,
      loading: false,
      organization: props.organization || '',
      unique: true
    };
    this.validateOrganization = debounce(this.validateOrganization, 500);
  }

  componentDidMount() {
    this.mounted = true;
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  stopLoading = () => {
    if (this.mounted) {
      this.setState({ loading: false });
    }
  };

  validateOrganization = (organization: string) => {
    getOrganization(organization).then(
      response => {
        if (this.mounted) {
          this.setState({ unique: response == null });
        }
      },
      () => {}
    );
  };

  sanitizeOrganization = (organization: string) =>
    organization
      .toLowerCase()
      .replace(/[^a-z0-9-]/, '')
      .replace(/^-/, '');

  handleOrganizationChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const organization = this.sanitizeOrganization(event.target.value);
    this.setState({ organization });
    this.validateOrganization(organization);
  };

  handleOrganizationCreate = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    const { organization } = this.state;
    if (organization) {
      this.setState({ loading: true });
      createOrganization({ key: organization, name: organization }).then(() => {
        if (this.mounted) {
          this.setState({ done: true, loading: false });
          this.props.onDone(organization);
        }
      }, this.stopLoading);
    }
  };

  handleOrganizationDelete = () => {
    const { organization } = this.state;
    if (organization) {
      this.setState({ loading: true });
      deleteOrganization(organization).then(() => {
        if (this.mounted) {
          this.setState({ done: false, loading: false, organization: '' });
          this.props.onDelete();
        }
      }, this.stopLoading);
    }
  };

  render() {
    const { done, loading, organization, unique } = this.state;

    const valid = unique && organization.length >= 2;

    return done ? (
      <div>
        <span className="spacer-right text-middle">{organization}</span>
        {loading ? (
          <i className="spinner text-middle" />
        ) : (
          <DeleteButton className="button-small" onClick={this.handleOrganizationDelete} />
        )}
      </div>
    ) : (
      <form onSubmit={this.handleOrganizationCreate}>
        <input
          autoFocus={true}
          className="input-super-large spacer-right text-middle"
          maxLength={32}
          minLength={2}
          onChange={this.handleOrganizationChange}
          placeholder={translate('onboarding.organization.placeholder')}
          required={true}
          type="text"
          value={organization}
        />
        {loading ? (
          <i className="spinner text-middle" />
        ) : (
          <SubmitButton className="text-middle" disabled={!valid}>
            {translate('create')}
          </SubmitButton>
        )}
        {!unique && (
          <span className="big-spacer-left text-danger text-middle">
            <AlertErrorIcon className="little-spacer-right text-text-top" />
            {translate('this_name_is_already_taken')}
          </span>
        )}
        <div className="note spacer-top abs-width-300">
          {translate('onboarding.organization.key_requirement')}
        </div>
      </form>
    );
  }
}
