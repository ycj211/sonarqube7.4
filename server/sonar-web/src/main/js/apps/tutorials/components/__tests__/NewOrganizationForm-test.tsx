/*
 * SonarQube
 * Copyright (C) 2009-2018 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { mount } from 'enzyme';
import NewOrganizationForm from '../NewOrganizationForm';
import { change, submit, waitAndUpdate } from '../../../../helpers/testUtils';

jest.mock('../../../../api/organizations', () => ({
  createOrganization: () => Promise.resolve(),
  deleteOrganization: () => Promise.resolve(),
  getOrganization: () => Promise.resolve(null)
}));

jest.mock('../../../../components/icons-components/ClearIcon');

it('creates new organization', async () => {
  const onDone = jest.fn();
  const wrapper = mount(<NewOrganizationForm onDelete={jest.fn()} onDone={onDone} />);
  expect(wrapper).toMatchSnapshot();
  change(wrapper.find('input'), 'foo');
  submit(wrapper.find('form'));
  expect(wrapper).toMatchSnapshot(); // spinner
  await waitAndUpdate(wrapper);
  expect(wrapper).toMatchSnapshot();
  expect(onDone).toBeCalledWith('foo');
});

it('deletes organization', async () => {
  const onDelete = jest.fn();
  const wrapper = mount(<NewOrganizationForm onDelete={onDelete} onDone={jest.fn()} />);
  wrapper.setState({ done: true, loading: false, organization: 'foo' });
  expect(wrapper).toMatchSnapshot();
  (wrapper.find('DeleteButton').prop('onClick') as Function)();
  wrapper.update();
  expect(wrapper).toMatchSnapshot(); // spinner
  await waitAndUpdate(wrapper);
  expect(wrapper).toMatchSnapshot();
  expect(onDelete).toBeCalled();
});
