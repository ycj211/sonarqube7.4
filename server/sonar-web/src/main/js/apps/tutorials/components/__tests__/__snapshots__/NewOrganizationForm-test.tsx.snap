// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates new organization 1`] = `
<NewOrganizationForm
  onDelete={[MockFunction]}
  onDone={[MockFunction]}
>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      className="input-super-large spacer-right text-middle"
      maxLength={32}
      minLength={2}
      onChange={[Function]}
      placeholder="onboarding.organization.placeholder"
      required={true}
      type="text"
      value=""
    />
    <SubmitButton
      className="text-middle"
      disabled={true}
    >
      <Button
        className="text-middle"
        disabled={true}
        preventDefault={false}
        type="submit"
      >
        <button
          className="button text-middle"
          disabled={true}
          onClick={[Function]}
          type="submit"
        >
          create
        </button>
      </Button>
    </SubmitButton>
    <div
      className="note spacer-top abs-width-300"
    >
      onboarding.organization.key_requirement
    </div>
  </form>
</NewOrganizationForm>
`;

exports[`creates new organization 2`] = `
<NewOrganizationForm
  onDelete={[MockFunction]}
  onDone={[MockFunction]}
>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      className="input-super-large spacer-right text-middle"
      maxLength={32}
      minLength={2}
      onChange={[Function]}
      placeholder="onboarding.organization.placeholder"
      required={true}
      type="text"
      value="foo"
    />
    <i
      className="spinner text-middle"
    />
    <div
      className="note spacer-top abs-width-300"
    >
      onboarding.organization.key_requirement
    </div>
  </form>
</NewOrganizationForm>
`;

exports[`creates new organization 3`] = `
<NewOrganizationForm
  onDelete={[MockFunction]}
  onDone={
    [MockFunction] {
      "calls": Array [
        Array [
          "foo",
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
>
  <div>
    <span
      className="spacer-right text-middle"
    >
      foo
    </span>
    <DeleteButton
      className="button-small"
      onClick={[Function]}
    >
      <ButtonIcon
        className="button-small"
        color="#d4333f"
        onClick={[Function]}
      >
        <Button
          className="button-small button-icon"
          onClick={[Function]}
          stopPropagation={true}
          style={
            Object {
              "color": "#d4333f",
            }
          }
        >
          <button
            className="button button-small button-icon"
            onClick={[Function]}
            style={
              Object {
                "color": "#d4333f",
              }
            }
            type="button"
          >
            <ClearIcon />
          </button>
        </Button>
      </ButtonIcon>
    </DeleteButton>
  </div>
</NewOrganizationForm>
`;

exports[`deletes organization 1`] = `
<NewOrganizationForm
  onDelete={[MockFunction]}
  onDone={[MockFunction]}
>
  <div>
    <span
      className="spacer-right text-middle"
    >
      foo
    </span>
    <DeleteButton
      className="button-small"
      onClick={[Function]}
    >
      <ButtonIcon
        className="button-small"
        color="#d4333f"
        onClick={[Function]}
      >
        <Button
          className="button-small button-icon"
          onClick={[Function]}
          stopPropagation={true}
          style={
            Object {
              "color": "#d4333f",
            }
          }
        >
          <button
            className="button button-small button-icon"
            onClick={[Function]}
            style={
              Object {
                "color": "#d4333f",
              }
            }
            type="button"
          >
            <ClearIcon />
          </button>
        </Button>
      </ButtonIcon>
    </DeleteButton>
  </div>
</NewOrganizationForm>
`;

exports[`deletes organization 2`] = `
<NewOrganizationForm
  onDelete={[MockFunction]}
  onDone={[MockFunction]}
>
  <div>
    <span
      className="spacer-right text-middle"
    >
      foo
    </span>
    <i
      className="spinner text-middle"
    />
  </div>
</NewOrganizationForm>
`;

exports[`deletes organization 3`] = `
<NewOrganizationForm
  onDelete={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  onDone={[MockFunction]}
>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      className="input-super-large spacer-right text-middle"
      maxLength={32}
      minLength={2}
      onChange={[Function]}
      placeholder="onboarding.organization.placeholder"
      required={true}
      type="text"
      value=""
    />
    <SubmitButton
      className="text-middle"
      disabled={true}
    >
      <Button
        className="text-middle"
        disabled={true}
        preventDefault={false}
        type="submit"
      >
        <button
          className="button text-middle"
          disabled={true}
          onClick={[Function]}
          type="submit"
        >
          create
        </button>
      </Button>
    </SubmitButton>
    <div
      className="note spacer-top abs-width-300"
    >
      onboarding.organization.key_requirement
    </div>
  </form>
</NewOrganizationForm>
`;
